frontend/src/app/App.routes.js

```diff
@@ -0,0 +1,17 @@
+import { removeLastSlash } from "../api/utils/api";
+
+export const BASE_ROUTE = "/";
+const SONGS_BASE_ROUTE = "/songs/";
+export const SONGS_EDIT_ROUTE = SONGS_BASE_ROUTE + "edit/";
+export const SONGS_CREATE_ROUTE = SONGS_BASE_ROUTE + "create/";
+
+export const navHome = history => history.push(removeLastSlash(BASE_ROUTE));
+
+export const navViewSong = (history, song_id) =>
+    history.push(removeLastSlash(BASE_ROUTE + song_id));
+
+export const navEditSong = (history, song_id) =>
+    history.push(removeLastSlash(SONGS_EDIT_ROUTE + song_id));
+
+export const navCreateSong = history =>
+    history.push(removeLastSlash(SONGS_CREATE_ROUTE));
```

Comment:
Any specific reason why this way is bad?

Reply:
How should I only perform the navigation if a request succeeds using `<a></a>`

Reply:
All times I use these javascript is necessary because they are either something provided to ReactDigitComponents or the navigations is conditional based on the success of a request or something of the like

Reply:
Use `Link` from `react-router-dom`

Reply:
Screen readers like to know what is clickable. Also you can do a CTRL + Click this way

Reply:
I'm a bit curious, at least for the create navigation it spontaneously feels to me like it probably shouldn't be a link. 
Partly because it might give an error and thus should not redirect but also because I wouldn't expect to be able to CTRL + click on the save button on such a page. But that might just be me. :)

Reply:
The other two however should probably be links.

Reply:
The dialog lies outside the root element and is therefore outside the router. So Link can't be used in a DigitDialog as I have to wrap the link in a new Router which destroys the navigation.

Reply:
I have converted to use Link where it isn't conditional on whether a request is successful or it isn't in a  DigitDialog.